subsys/shell/shell_ops.c

```diff
@@ -5,24 +5,42 @@
  */
 
 #include <ctype.h>
+#include <stdio.h>
 #include "shell_ops.h"
 
+#define CMD_CURSOR_LEN 8
 void z_shell_op_cursor_vert_move(const struct shell *shell, int32_t delta)
 {
-	if (delta != 0) {
-		z_shell_raw_fprintf(shell->fprintf_ctx, "\033[%d%c",
-				    delta > 0 ? delta : -delta,
-				    delta > 0 ? 'A' : 'B');
+	uint8_t cmd[CMD_CURSOR_LEN];
+	char dir = delta > 0 ? 'A' : 'B';
+
+	if (delta == 0) {
+		return;
+	}
+
+	if (delta < 0) {
+		delta = -delta;
 	}
+
+	sprintf(cmd, "\e[%d%c", delta, dir);
+	Z_SHELL_VT100_CMD(shell, cmd);
 }
 
 void z_shell_op_cursor_horiz_move(const struct shell *shell, int32_t delta)
 {
-	if (delta != 0) {
-		z_shell_raw_fprintf(shell->fprintf_ctx, "\033[%d%c",
-				    delta > 0 ? delta : -delta,
-				    delta > 0 ? 'C' : 'D');
+	uint8_t cmd[CMD_CURSOR_LEN];
+	char dir = delta > 0 ? 'C' : 'D';
+
+	if (delta == 0) {
+		return;
 	}
+
+	if (delta < 0) {
+		delta = -delta;
+	}
+
+	sprintf(cmd, "\e[%d%c", delta, dir);
+	Z_SHELL_VT100_CMD(shell, cmd);
```

Comment:
with small modifications of Z_SHELL_VT100_CMD  it could be turned into:
`Z_SHELL_VT100_CMD(shell, "\e[%d%c", delta, dir);`